src/geopackage/read.cpp

```diff
@@ -0,0 +1,108 @@
+#include "GeoPackage.hpp"
+
+#include <numeric>
+
+std::shared_ptr<geojson::FeatureCollection> geopackage::read(
```

Comment:
might be a good idea to add some tests, especially with the id subsetting...

Reply:
6a0bb77 adds a test for the subsetting. There already is a test for reading a GeoPackage using this function (and for projection), but without adding more GeoPackage files, there isn't much more to test (as far as I can think of). Since the output is a pointer to a `geojson::FeatureCollection`, the tests for that are handled within the geojson tests.